### PR TITLE
Print hooks as they appear in spec reporter

### LINKED ISSUE #4571: Hook errors are printed in the end in spec reporter for Jasmine/Cucumber frameworks
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.14.3

**Describe the bug**
Errors in before/after All hooks are printed in the end.

**To Reproduce**
1. throw an error in BeforeAll hook
2. see error in spec reporter

**Expected behavior**
tests/hooks should be printed in chronological order

---

image

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

Print hooks as they appear in spec reporter for Jasmine and Cucumber. Mocha is not affected.

Added combined list of hooks/tests to store them in order as they appear.

image
image

fixes #4571 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works

## Further comments

ref https://github.com/webdriverio/webdriverio/pull/4572 https://github.com/webdriverio/webdriverio/pull/4566

@martinfrancois

### CODE REVIEW

**Comment by mgrybyk** [comment]
> not the most elegant

100% agree. Failed to find a better naming as well.
It is an array that combines specs and hooks that are added as they appear in chronological order. Just wrote this sentence to help imagine a better naming

**Comment by martinfrancois** [comment]
> Failed to find a better naming as well.

I thought so, since I was already thinking about that when that combined array popped up in the discussion 😄 

To bring some further "inspiration" for the name: It would be great if it would be instantly recognizable by people implementing reporters that this would be the right one to use if they're thinking about which one to use to show the "steps" in a report. Some keywords I was thinking about: event, stat, sequence, stack (in case that helps anyone)

File: packages/wdio-reporter/src/index.js [hunk 1]
```diff
@@ -131,8 +133,10 @@ export default class WDIOReporter extends EventEmitter {
             const suiteTests = currentSuite.tests
             if (!suiteTests.length || currentTest.uid !== suiteTests[suiteTests.length - 1].uid) {
                 currentSuite.tests.push(currentTest)
+                currentSuite.hooksAndTests.push(currentTest)
             } else {
                 suiteTests[suiteTests.length - 1] = currentTest
+                currentSuite.hooksAndTests[ currentSuite.hooksAndTests.length - 1] = currentTest
```
Comment (CrispusDH) [inline packages/wdio-reporter/src/index.js]: here one extra space after `[`

Comment (christian-bromann) [inline packages/wdio-reporter/src/index.js]:
```suggestion
                currentSuite.hooksAndTests[currentSuite.hooksAndTests.length - 1] = currentTest
```

**Pull request merged**